nautobot/docs/release-notes/version-2.4.md

```diff
@@ -8,13 +8,114 @@ This document describes all new features and changes in Nautobot 2.4.
+
+[Adoption of this framework](../development/apps/migration/ui-component-framework/index.md) significantly reduces the amount of boilerplate needed to define an object detail view, drives increased self-consistency across views, and encourages code reuse. It also insulates Apps from the details of Nautobot's CSS (Bootstrap 3 framework), smoothing the way for Nautobot to adopt a new CSS framework in the future with minimal impact to compliant apps.
+
+App [template extensions](../development/apps/api/ui-extensions/object-views.md) of Nautobot core views can also be reimplemented using the UI Component Framework and are recommended to do so.
+
+As of Nautobot 2.4.0, the following detail views have been migrated to use the UI Component Framework:
+
+- Circuit
+- Cluster Type
+- Device ("Add Components" buttons only)
+- External Integration
+- Location Type
+- Provider
+- Route Target
+- Secret
+- Tenant
+- VRF
+
+#### Virtual Device Context Data Models
+
+Nautobot 2.4 adds a [`VirtualDeviceContext`](../user-guide/core-data-model/dcim/virtualdevicecontext.md) data model to support modeling of logical partitions of physical network devices, such as Cisco Nexus Virtual Device Contexts, Juniper Logical Systems, Arista Multi-instance EOS, and so forth. Device Interfaces can be associated to Virtual Device Contexts via the new `InterfaceVDCAssignment` model as well.
+
+#### Wireless Data Models
+
+Nautobot 2.4 adds the data models [`WirelessNetwork`](../user-guide/core-data-model/wireless/wirelessnetwork.md), [`RadioProfile`](../user-guide/core-data-model/wireless/radioprofile.md), and [`SupportedDataRate`](../user-guide/core-data-model/wireless/supporteddatarate.md), enabling Nautobot to model campus wireless networks. In support of this functionality, the [`Controller`](../user-guide/core-data-model/dcim/controller.md) and [`ControllerManagedDeviceGroup`](../user-guide/core-data-model/dcim/controllermanageddevicegroup.md) models have been enhanced with additional capabilities as well.
+
+Refer to the [Wireless](../user-guide/core-data-model/wireless/index.md) documentation for more details.
+
+### Deprecated
+
+#### `FilterTestCases.NameOnlyFilterTestCase` and `FilterTestCases.NameSlugFilterTestCase`
+
+These two generic base test classes are deprecated. Apps should migrate to using `FilterTestCases.FilterTestCase` with an appropriately defined list of `generic_filter_tests` instead.
+
+#### Job `task_queues` and `ScheduledJob.queue`
+
+The `Job.task_queues` field (a list of queue name strings) is deprecated in favor of the new `Job.job_queues` relationship to the `JobQueue` model. `task_queues` is still readable and settable for backward compatibility purposes but code using this attribute should migrate to using `job_queues` instead.
```

Comment:
Should we include `default_job_queue` in some regard here as well?

Reply:
This is the "deprecated" section, is there some existing functionality that was deprecated in favor of `default_job_queue`?

Reply:
I think since job_queues are not sorted/user specified anymore, we should mention the fact that the first entry of the legacy `task_queues`(list of queue name strings) will be replaced now by `default_job_queue`. Definitely not a blocker.

Reply:
I'll add a note in the `Job Authors & App Developers` section above to this effect.